File: src/main/java/com/example/ui/pages/heroku/DragDropPage.java
package com.example.ui.pages.herokuapp;

import com.example.base.TestContext;
import com.example.ui.pages.BasePage;
import com.microsoft.playwright.Page;
import lombok.extern.slf4j.Slf4j;

/**
 * Drag and Drop page ({@code /drag_and_drop}).
 */
@Slf4j
public class DragDropPage extends BasePage {

    public DragDropPage(Page page, TestContext testContext) {
        super(page, testContext);
    }

    public void dragAToB() {
        page.locator("#column-a").dragTo(page.locator("#column-b"));
    }

    public String columnHeader(String columnId) {
        return getText(page.locator("#" + columnId + " header"), columnId + " header").trim();
    }
}

File: typescript/src/pages/heroku/drag-drop.page.ts
import { BasePage } from '../base.page';

/** Drag and Drop page (/drag_and_drop). */
export class DragDropPage extends BasePage {
  async dragAToB(): Promise<void> {
    await this.page.locator('#column-a').dragTo(this.page.locator('#column-b'));
  }

  async columnHeader(columnId: 'column-a' | 'column-b'): Promise<string> {
    return ((await this.page.locator(`#${columnId} header`).innerText()) ?? '').trim();
  }
}
